Realizado registro de usuarios y tests-#99

alfgonfer opened the pull request:
Se ha realizado el backend de registro de usuarios y una serie de tests para probar los distintos casos: la contraseña debe coincidir, no puede haber dos usuarios con el mismo username y no puede haber dos usuarios con el mismo correo.
La integración de backend-frontend se realizará en la rama correspondiente al frontend de registro.

alfgonfer committed:
Realizado registro de usuarios y tests-#99

Se ha realizado el backend de registro de usuarios y una serie de tests para probar los distintos casos: la contraseña debe coincidir, no puede haber dos usuarios con el mismo username y no puede haber dos usuarios con el mismo correo.

jorandmol reviewed (approved):
Veo todo correcto, aunque vamos a implementar la vista de registro dentro del login.html. Habría que mezclar los controladores de registro para que a la vista de login se envíen ambos formularios

jorandmol reviewed (changes requested):
Habría que mezclar los controladores de registro y login para que sean uno solo. De esta forma integramos las dos funcionalidades en la vista de login, haciendo que quede más limpio.

jorandmol reviewed (commented):
Cambio de opinión; está todo bien. Únicamente necesitaría que se enviase el listado de usernames para validar en la interfaz que el cliente no está introduciendo uno ya existente.

fragonlug committed:
Añadido la lista de usuarios y la validación-#99

Se ha realizado los ajustes pertinentes requeridos por el revisor, mandando la lista de usuarios para su validación. A su vez, se ha añadido la validación de los campos en el backend. Por otro lado se ha eliminado un print residual de commits previos.

alfgonfer committed:
Redireccionado de registro con errores funcionando-#99

Se ha corregido un pequeño error en el views.py y se ha añidido un html muy simple para pasar las pruebas. Queda pendiente integración con frontend.

mrf1989 reviewed (changes requested):
Se os ha colado el registro.html en el commit. Lo suyo sería dejarlo fuera para que no entre en conflicto con el trabajo de @jorandmol

jorandmol commented:
> Se os ha colado el registro.html en el commit. Lo suyo sería dejarlo fuera para que no entre en conflicto con el trabajo de @jorandmol

Parece que es por la necesidad de que pasen algunos test. No debe de haber problema ya que está controlando.

Una vez se añadan estos cambios en la rama develop me encargaré de integrar frontend y backend en la rama de la tarea #98

fragonlug commented:
> Se os ha colado el registro.html en el commit. Lo suyo sería dejarlo fuera para que no entre en conflicto con el trabajo de @jorandmol

Es necesario que exista para la realización de los test del registro, se espera que cuando se integre con el frontend, sea sustituido por completo, teniendo así 0 impacto sobre el trabajo realizado de @jorandmol .
